src/argilla/server/schemas/v1/records.py

```diff
@@ -51,3 +51,11 @@ class RecordUpdate(UpdateSchema):
     metadata_: Optional[Dict[str, Any]] = Field(None, alias="metadata")
     suggestions: Optional[List[SuggestionCreate]] = None
     vectors: Optional[Dict[str, List[float]]]
+
+    @validator("metadata_", pre=True)
+    @classmethod
+    def skip_nan_values(cls, metadata: Optional[Dict[str, Any]]) -> Optional[Dict[str, Any]]:
+        if metadata is None:
+            return metadata
+
+        return {k: v for k, v in metadata.items() if v == v}  # By definition, NaN != NaN
```

Comment:
Not that the comment from @jfcalvo applies here too https://github.com/argilla-io/argilla/pull/4477/files#r1448542743

Reply:
ditto